src/main/resources/assets/js/app/installation/view/MarketAppsTreeGrid.ts

```diff
@@ -256,26 +255,40 @@ export class MarketAppsTreeGrid
 
     private updateAppsStatuses(applications: MarketApplication[]): void {
         const marketApps = applications.filter(marketApp => !!marketApp.getLatestVersion());
-        const installedApps = this.installedApplications.slice();
-        const installedAppsIds = installedApps.map(app => app.getApplicationKey().toString());
-        const getInstalledAppIndex = (marketApp: MarketApplication) => installedAppsIds.indexOf(marketApp.getAppKey().toString());
 
         marketApps.forEach(marketApp => {
-            const installedAppIndex = getInstalledAppIndex(marketApp);
-            const isMarketAppInstalled = installedAppIndex >= 0;
-            if (isMarketAppInstalled) {
-                if (MarketAppsTreeGridHelper.installedAppCanBeUpdated(marketApp, installedApps[installedAppIndex])) {
-                    marketApp.setStatus(MarketAppStatus.OLDER_VERSION_INSTALLED);
-                } else {
-                    marketApp.setStatus(MarketAppStatus.INSTALLED);
-                }
-            } else if (marketApp.getStatus() === MarketAppStatus.INSTALLED ||
-                       marketApp.getStatus() === MarketAppStatus.OLDER_VERSION_INSTALLED) {
-                marketApp.setStatus(MarketAppStatus.NOT_INSTALLED);
-            }
+            const installedApp = this.getInstalledAppByMarketApp(marketApp);
+            marketApp.setStatus(this.getAppStatusForInstallDialog(marketApp, installedApp));
         });
     }
 
+    private getAppStatusForInstallDialog(marketApp: MarketApplication, installedApp?: Application): MarketAppStatus {
+        if (installedApp) {
+            if (MarketAppsTreeGridHelper.installedAppCanBeUpdated(marketApp, installedApp)) {
+               return MarketAppStatus.OLDER_VERSION_INSTALLED;
+            }
+
+            return MarketAppStatus.INSTALLED;
+        }
+
+        if (marketApp.getStatus() === MarketAppStatus.INSTALLED ||
+            marketApp.getStatus() === MarketAppStatus.OLDER_VERSION_INSTALLED) {
+            return MarketAppStatus.NOT_INSTALLED;
+        }
+
+        return marketApp.getStatus();
+    }
+
+    private getInstalledAppByMarketApp(marketApp: MarketApplication): Application | undefined {
+        return this.installedApplications.find(
+            (app: Application) => app.getApplicationKey().equals(marketApp.getAppKey()) || app.getApplicationKey().toString() ===
+                                  this.replaceDashesWithDotsInMarketApp(marketApp));
```

Comment:
This will call the `this.replaceDashesWithDotsInMarketApp` for every item in `this.installedApplications`, even though `marketApp` is the same. Call it once before the loop, then compare with the calculated value inside the loop.